skrf/media/media.py

```diff
@@ -1310,6 +1310,55 @@ def shunt_inductor(self, L: NumberLike, **kwargs) -> Network:
         return self.shunt(self.inductor(L=L, **kwargs) **
                           self.short(**kwargs), **kwargs)
 
+    def capacitor_q(self, C: NumberLike, f: NumberLike, q_factor: NumberLike, **kwargs) -> Network:
+        r"""
+        Capacitor with Q factor.
+
+        Parameters
+        ----------
+        C : number, array-like
+            Capacitance in Farads.
+        f : number
+            Frequency at which Q is defined
+        q_factor : number
+            Q-factor of capacitor
+        \*\*kwargs : arguments, keyword arguments
+            passed to func:`capacitor`
+
+        Returns
+        -------
+        capacitor_q : :class:`~skrf.network.Network` object
+            capacitor_q (2-port)
+
+        """
+        idea_cap = self.shunt(self.capacitor(C=C))
+        rac = q_factor / (C * 2 * npy.pi * f)
+        idea_res = self.shunt(self.resistor(R=rac))
+
+        return innerconnect(connect(idea_cap, 1, idea_res, 2), 1, 3)
+
+    def inductor_q(self, L: NumberLike, q_factor: NumberLike, **kwargs) -> Network:
+        r"""
+        Inductor with Q factor.
+
+        Parameters
+        ----------
+        L : number, array-like
+            Inductance in Henries.
+        q_factor : number
+            Q-factor of inductor
+        \*\*kwargs : arguments, keyword arguments
+            passed to func:`inductor`
+
+        Returns
+        -------
+        inductor_q : :class:`~skrf.network.Network` object
+            inductor_q (2-port)
+        
+        """
+        rac = self.frequency.w * L / q_factor
```

Comment:
I don't understand : Here the frequency of the object (`self.frequency`) is used instead of a constant like in the `capacitor_q`. `self.frequency.f` is an array, so will be `rac` and the resistor will be frequency dependant (at the contrary of `capacitor_q`).

Reply:
Yes, you are right. I also had this problem when reproducing the s2p files of INDQ and CAPQ generated under the default configuration of ADS.
By searching the manual and comparing the s2p files, I found that the resistance of the {Inductor with Q} in ADS is indeed proportional to the frequency, and the conductance of the {Capacitor with Q} are constant by default.
The following are the results of the test:

First is the Capacitor with Q generate from ADS (freq_start=1GHz, freq_stop=3GHz, F0=2GHz, Q=30, C=1pF), and the result is that the snp generated by skrf is exactly the same as the snp generated by ADS.
```
import skrf as rf
import numpy as np

ntwk_std = rf.Network('ADS_CQ.s2p')
line = rf.media.DefinedGammaZ0(ntwk_std.frequency)
ntwk_test = line.capacitor_q(C=1e-12, q_factor=30, f_0=2e9)
np.allclose(ntwk_test.s, ntwk_std.s)
True
```
The second test is the Inductor with Q generate from ADS (freq_start=1GHz, freq_stop=3GHz, F0=2GHz, Q=30, L=1nH), and the result is that the snp generated by skrf has an error about 1e-6 with the snp generated by ADS, but I don't know the source of the error.
```
import skrf as rf
import numpy as np

ntwk_std = rf.Network('ADS_LQ.s2p')
line = rf.media.DefinedGammaZ0(ntwk_std.frequency)
ntwk_test = line.inductor_q(L=1e-9, q_factor=30)
>>> ntwk_test.s
array([[[0.0059978 +0.06232447j, 0.9940022 -0.06232447j],
        [0.9940022 -0.06232447j, 0.0059978 +0.06232447j]],

       [[0.00600765+0.06238604j, 0.99399235-0.06238604j],
        [0.99399235-0.06238604j, 0.00600765+0.06238604j]],

       [[0.0060175 +0.06244762j, 0.9939825 -0.06244762j],
        [0.9939825 -0.06244762j, 0.0060175 +0.06244762j]],

       ...,

       [[0.03988399+0.17972844j, 0.96011601-0.17972844j],
        [0.96011601-0.17972844j, 0.03988399+0.17972844j]],

       [[0.03990754+0.1797836j , 0.96009246-0.1797836j ],
        [0.96009246-0.1797836j , 0.03990754+0.1797836j ]],

       [[0.0399311 +0.17983876j, 0.9600689 -0.17983876j],
        [0.9600689 -0.17983876j, 0.0399311 +0.17983876j]]])
>>> ntwk_std.s
array([[[0.00600551+0.0623235j , 0.99399449-0.0623235j ],
        [0.99399449-0.0623235j , 0.00600551+0.0623235j ]],

       [[0.00601535+0.06238507j, 0.99398465-0.06238507j],
        [0.99398465-0.06238507j, 0.00601535+0.06238507j]],

       [[0.00602519+0.06244665j, 0.99397481-0.06244665j],
        [0.99397481-0.06244665j, 0.00602519+0.06244665j]],

       ...,

       [[0.03988012+0.17972994j, 0.96011988-0.17972994j],
        [0.96011988-0.17972994j, 0.03988012+0.17972994j]],

       [[0.03990366+0.17978511j, 0.96009634-0.17978511j],
        [0.96009634-0.17978511j, 0.03990366+0.17978511j]],

       [[0.03992722+0.17984027j, 0.96007278-0.17984027j],
        [0.96007278-0.17984027j, 0.03992722+0.17984027j]]])
```

[ADS_S2P.zip](https://github.com/scikit-rf/scikit-rf/files/12078278/ADS_S2P.zip)

Reply:
OK. The fact that the ref frequency is not done the same way is puzzling. If confirmed, it should be mentionned in the docstring.